src/abi.md

```diff
@@ -0,0 +1,61 @@
+# Application Binary Interface (ABI)
+
+This section documents features that affect the ABI of the compiled output of
+a crate.
+
+See *[extern functions]* for information on specifying the ABI for exporting
+functions. See *[external blocks]* for information on specifying the ABI for
+linking external libraries.
+
+## The `used` attribute
+
+The *`used` attribute* can only be applied to [`static` items]. This [attribute] forces the
+compiler to keep the variable in the output object file (.o, .rlib, etc.) even if the variable is
+not used, or referenced, by any other item in the crate.
+
+Below is an example that shows under what conditions the compiler keeps a `static` item in the
+output object file.
+
+``` rust
+// foo.rs
+
+// This is kept because of `#[used]`:
+#[used]
+static FOO: u32 = 0;
+
+// This is removable because it is unused:
+#[allow(dead_code)]
+static BAR: u32 = 0;
+
+// This is kept because it is referenced by a public, reachable function:
```

Comment:
This should be on `QUUX`, not `BAZ`. `BAZ` is kept because it is a public static.